bite-size-notes/week-7-a-kalman-filter.md

```diff
@@ -0,0 +1,129 @@
+
+## Preliminaries
+Given the following variables and initial conditions
+- $x$ is the true state
+- $y$ is the output
+- $u$ is the input
+- $\hat{x}$ is the estimated state
+- $\tilde{x}=x-\hat{x}$ is the estimation error between the true and estimated states
+- $K$ is the Kalman filter gain matrix
+- $A,B,C,D$ are the system state space matrices
+- $\omega\approx\mathcal{N}(0,Q)$ is some Gaussian process noise sampled from a normal distribution with variance $Q=\mathbb{E}[w_kw_k^T]$
+- $v\approx\mathcal{N}(0,R)$ is some Gaussian observer noise sampled from a normal distribution with variance $R=\mathbb{E}[v_kv_k^T]$
+- $\hat{x}_0\approx\mathcal{N}(\mu_0,\Sigma_0)$ is the initial state estimate
+
+The following linear discrete time Gaussian dynamics with some additive Gaussian noise will be used to represent a system of interest.
+
+$$x_{k+1}=Ax_k+Bu_k+\omega_k$$
+
+$$y_{k+1}=Cx_k+Du_k+v_k$$
+
+For the puposes of these notes, $D$ is assumed to be zero.
+
+Additionally, recall that passing a Gaussian through linear dynamics produces another Gaussian.
+
+## Main Body
+The Kalman filter is a method of estimating unknown variables in a system's current state. It is a recursive process that uses successive observations to update an estimate of the true state. This estimate is represented by a normal distribution with some mean $\mu$ (the average estimated state) and covariance $\Sigma$ (the relationships between state variables).
+### Kalman Filter Implementation
+The Kalman filter is a two-step process. Both steps must be run at each time step $k$.
+1) **Estimate the current state based on the previous state**
+   
+   $$\mu_k^{pred}=A\mu_{k-1}+Bu_{k-1}$$
+
+   $$\Sigma_k^{pred}=A\Sigma_{k-1}A^T+Q$$
+
+   Here the current predicted mean is estimated by passing the previous mean through the system dynamics. The estimation equation for the predicted variance is derived later in these notes and comprises the bulk of the Kalman filter derivation process.
+
+2) **Update the current state estimation using observations of the outputs**
+
+   The current optimal Kalman gain matrix $K_k$ is calculated using the current predicted variance.
+
+   $$K_k=\Sigma_{k}^{pred}C^T(C\Sigma_{k}^{pred}C^T+R)^{-1}$$
+
+   The Kalman gain is then used to correct the predicted mean and covariance using current observations of the state $y_k$.
+   
+   $$\mu_k=(I-K_kC)\mu_k^{pred}+K_ky_k$$
+
+   $$\Sigma_k=(I-K_kC)\Sigma_k^{pred}$$
+
+$\mu_k$ can then be used to represent the true state $x_k$k for the purposes of control.
+
+It is important to note that the Kalman filter process runs forwards in time.
+### Kalman Filter Derivation
```

Comment:
This section provides a detailed derivation of the Kalman filter equations, which is essential for a deep understanding.